source/MaterialXTest/MaterialXCore/Document.cpp

```diff
@@ -119,42 +119,94 @@ TEST_CASE("Version", "[document]")
     mx::FileSearchPath searchPath("resources/Materials/TestSuite/stdlib/upgrade/");
 
-    mx::readFromXmlFile(doc, "1_36_to_1_37.mtlx", searchPath);
-    REQUIRE(doc->validate());
+    // 1.36 to 1.37
+    {
+        mx::readFromXmlFile(doc, "1_36_to_1_37.mtlx", searchPath);
+        REQUIRE(doc->validate());
 
-    mx::XmlWriteOptions writeOptions;
-    writeOptions.writeXIncludeEnable = true;
-    mx::writeToXmlFile(doc, "1_36_to_1_37_updated.mtlx", &writeOptions);
+        mx::XmlWriteOptions writeOptions;
+        writeOptions.writeXIncludeEnable = true;
+        mx::writeToXmlFile(doc, "1_36_to_1_37_updated.mtlx", &writeOptions);
 
-    mx::DocumentPtr doc2 = mx::createDocument();
-    mx::readFromXmlFile(doc2, "1_36_to_1_37_updated.mtlx");
-    REQUIRE(doc2->validate());
+        mx::DocumentPtr doc2 = mx::createDocument();
+        mx::readFromXmlFile(doc2, "1_36_to_1_37_updated.mtlx");
+        REQUIRE(doc2->validate());
 
-    // Check conversion to desired types occurred
-    std::unordered_map<std::string, unsigned int> convertSet = 
-    {
-        { "invertmatrix", 2}, 
-        { "rotate2d", 1},
-        { "rotate3d", 1},
-        { "transformmatrix", 8},
-        { "ifgreatereq", 7}, 
-        { "separate2", 1}, 
-        { "separate3", 1},
-        { "separate4", 1},
-        { "combine2", 1}, 
-        { "combine3", 1},
-        { "combine4", 1}
-    };
-    for (mx::NodePtr node : doc2->getNodes())
-    {
-        auto convertItem = convertSet.find(node->getCategory());
-        if (convertItem != convertSet.end())
+        // Check conversion to desired types occurred
+        std::unordered_map<std::string, unsigned int> convertSet =
+        {
+            { "invertmatrix", 2},
+            { "rotate2d", 1},
+            { "rotate3d", 1},
+            { "transformmatrix", 8},
+            { "ifgreatereq", 7},
+            { "separate2", 1},
+            { "separate3", 1},
+            { "separate4", 1},
+            { "combine2", 1},
+            { "combine3", 1},
+            { "combine4", 1}
+        };
+        for (mx::NodePtr node : doc2->getNodes())
         {
-            convertItem->second--;
+            auto convertItem = convertSet.find(node->getCategory());
+            if (convertItem != convertSet.end())
+            {
+                convertItem->second--;
+            }
+        }
+        for (auto convertItem : convertSet)
+        {
+            REQUIRE((convertItem.second == 0));
         }
     }
-    for (auto convertItem : convertSet)
+
+    // 1.37 to 1.38
     {
-        REQUIRE((convertItem.second == 0));
+        mx::XmlReadOptions options;
+        options.desiredMajorVersion = 1;
+        options.desiredMajorVersion = 38;
+        doc = mx::createDocument();
+        mx::loadLibrary(mx::FilePath::getCurrentPath() / mx::FilePath("libraries/stdlib/stdlib_defs.mtlx"), doc, nullptr, &options);
```

Comment:
Small extension on the utility to allow read options to be sent over to allow upgrading the definitions.